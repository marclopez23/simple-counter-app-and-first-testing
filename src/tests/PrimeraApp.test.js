import React from "react";
//import { render } from "@testing-library/react";
import { shallow } from "enzyme";
import PrimeraApp from "../Views/PrimeraApp";

describe("Pruebas en PrimeraApp", () => {
  /*test("debe de mostrar el mensaje", () => {
    const saludo = "Hola soy Marc";
    const { getByText } = render(<PrimeraApp saludo="Hola soy Marc" />);

    expect(getByText(saludo)).toBeInTheDocument();
  });*/

  test("debe de mostar PrimeraApp", () => {
    const saludo = "Hola soy Marc";
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostar el subtitulo que le pasamos por props", () => {
    const saludo = "Hola soy Marc";
    const subtitulo = "Frontend Developer";
    const wrapper = shallow(
      <PrimeraApp saludo={saludo} subtitulo={subtitulo} />
    );
    const txtParrafo = wrapper.find("p").text();
    expect(txtParrafo).toBe(subtitulo);
  });
});
