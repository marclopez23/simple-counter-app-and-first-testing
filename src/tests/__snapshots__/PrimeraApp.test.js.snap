// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en PrimeraApp debe de mostar PrimeraApp 1`] = `
<Fragment>
  <h1>
    Hola soy Marc
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
